limitador/src/counter.rs

```diff
@@ -48,6 +48,17 @@ impl Counter {
         self.limit.max_value()
     }
 
+    pub fn update_to_limit(&mut self, limit: &Limit) -> bool {
```

Comment:
I wonder if it makes sense to get rid of the limit included in the counter. This `update_to_limit` is solely to keep in sync limit objects representing the same. And it is the source of panic's because limits are not equal and should be.